src/dedup.rs

```diff
@@ -109,7 +114,7 @@ pub async fn dedup_img(
             .await?;
 
         if let Some(record) = &record {
-            return Ok(Some(record.message_id));
+            return Ok(Some((record.message_id, record.duplicate_cnt)));
         }
 
         if record.map_or(true, |record| record.dist != 0) {
```

Comment:
You might want to modify `insert_img.sql` as well because image entities can't have their duplicate count updated now.

There was a logic bug in the code, which prevented new images from being added to the database if a similar image already existed. The bug has been fixed in the latest master branch, so be sure to check it out when rebasing.